### PR TITLE
Related to BZ #1183802 - check `puppetrun!` method status

### PR DESCRIPTION
**Pull request by jistr**
https://bugzilla.redhat.com/show_bug.cgi?id=1183802

This will not solve the bug but might help us see that something went
wrong, but only if `puppetrun!` method actually returns false when
BZ#1183802 occurs.

### CODE REVIEW

**Comment by knowncitizen** [comment]
ACK

File: app/lib/actions/staypuft/host/puppet_run.rb [hunk 1]
```diff
@@ -23,8 +23,23 @@ def plan(host)
...
+          # try puppetrun 3 times
+          result = false
+          tries = 0
+          while !result && tries < 3
+            result = ::Host.find(host_id).puppetrun!
+            tries += 1
+          end
+
+          unless result
+            fail(::Staypuft::Exception, "Puppet run failed for host: #{host_id}")
```
Comment (iNecas) [inline app/lib/actions/staypuft/host/puppet_run.rb]: You might want to save the host.error_messages to get more details about the error into the action itself

Comment (jistr) [inline app/lib/actions/staypuft/host/puppet_run.rb]: If i put something to output[:error_message] and than run the `fail()` method, will the output be saved?

Comment (iNecas) [inline app/lib/actions/staypuft/host/puppet_run.rb]: It should be

Comment (jistr) [inline app/lib/actions/staypuft/host/puppet_run.rb]: Ack, thanks :)

**Comment by jistr** [comment]
Tested -- worked fine. I didn't hit the BZ case but the "everything is ok" case went through no problem.

**Comment by sseago** [comment]
ACK. merging

**Pull request merged**